src/header/header.scss

```diff
@@ -0,0 +1,168 @@
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import '../user-icon/index';
+@import '../icon/index';
+
+@mixin iui-header {
+  @include iui-reset;
+  @include iui-font-family;
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+  height: $iui-baseline * 5;
+  transition: height $iui-speed-fast ease-out;
+  @include themed {
+    background-color: t(iui-color-background-1);
+    border-bottom: $iui-xxs solid t(iui-color-background-5);
+    color: t(iui-text-color);
+  }
+
+  > .iui-left,
+  > .iui-center,
+  > .iui-right {
+    display: flex;
+    align-items: center;
+    height: 100%;
+  }
+
+  > .iui-left {
+    flex: 2 1 33%;
+    @include iui-header-buttons;
+
+    > .iui-chevron {
+      @include iui-icons-small;
+      flex-shrink: 0;
+      margin: 0 $iui-xs;
+    }
+  }
+
+  > .iui-center {
+    justify-content: center;
+    flex: 1 1 auto;
+  }
+
+  > .iui-right {
+    flex: 2 1 33%;
+    justify-content: flex-end;
+
+    .iui-user-icon {
+      transition: width $iui-speed-fast ease-out,
+        height $iui-speed-fast ease-out, font-size $iui-speed-fast ease-out;
+
+      * {
+        transition: width $iui-speed-fast ease-out,
+          height $iui-speed-fast ease-out, font-size $iui-speed-fast ease-out,
+          line-height $iui-speed-fast ease-out,
+          font-weight $iui-speed-fast ease-out;
+      }
+    }
+  }
+
+  .iui-button {
+    height: 100%;
+    border-radius: 0;
+  }
+
+  .iui-divider {
+    width: 1px;
+    height: $iui-baseline * 2;
+    @include themed {
+      background-color: t(iui-color-background-4);
+    }
+  }
+
+  &.iui-slim {
+    height: round($iui-baseline * 2.5);
+
+    > .iui-left {
+      .iui-header-button {
+        &:first-of-type {
+          .iui-header-button-icon {
+            margin: 0 $iui-m;
+          }
+
+          > .iui-label {
+            font-size: $iui-font-size;
+          }
+        }
+
+        .iui-header-button-icon {
+          width: $iui-icons-default;
+          height: $iui-icons-default;
+        }
+
+        .iui-description {
+          display: none;
+        }
+      }
+    }
+
+    > .iui-right {
+      .iui-user-icon {
+        @include iui-user-icon-size(
+          $size: $iui-icons-large,
+          $font-size: $iui-font-size-small
+        );
+      }
+    }
+  }
+}
+
+@mixin iui-header-button-icon(
+  $size: $iui-icons-large,
+  $border-radius: $iui-border-radius
+) {
+  width: $size;
+  height: $size;
+  border-radius: $border-radius;
+}
+
+@mixin iui-header-buttons {
+  .iui-header-button {
+    > .iui-label {
+      text-align: left;
+      font-size: $iui-font-size;
+
+      .iui-description {
+        @include themed {
+          opacity: t(iui-opacity-4);
+        }
+      }
+    }
+
+    .iui-header-button-icon {
+      @include iui-header-button-icon;
+    }
+
+    // For application button
+    &:first-of-type {
```

Comment:
Why not to be explicit and have a class name? Because we allow to pass custom classes in React.

Reply:
Well, it's designed to use the same component

Reply:
So that means that all of these require click handlers as we cant type them with div. (otherwise, the breadcrumb will become first of type) is that what we expect ?

Reply:
Like we discussed on wednesday, a `div` will not allow to tab into it and get focus, so we might want to do that if there is no click handlers on the application button. however, if we do that while we use the "first-of-type" to control the application button style, then the application button will get the type, but then the first breadcrumb will also get it. (first of type is based on the div/button, not on the class.)

Reply:
> _first of type is based on the div/button, not on the class_
Oh... I didn't know that. Hmm, that changes things 🤔 Maybe we do need a new class then. Will explore on Monday

Reply:
Added `.iui-header-logo` class which needs to be set on a div. To get focus and hover effects, add `type=button` and `tabindex=0` to div.

Reply:
I've changed to `role=button`, which is good but it needs a manual keydown listener.

See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#required_javascript_features